paywall/src/components/lock/Overlay.js

```diff
@@ -45,43 +55,106 @@ export const Overlay = ({
   showModal,
   scrollPosition,
   openInNewWindow,
-  config: { isInIframe },
-}) => (
-  <FullPage>
-    <Banner scrollPosition={scrollPosition} data-testid="paywall-banner">
-      <Headline id="Paywall_Headline">
-        You have reached your limit of free articles. Please purchase access
-      </Headline>
-      <Locks>
-        <GlobalErrorConsumer displayError={displayError(!isInIframe)}>
-          {locks.map(lock => (
-            <Lock
-              key={JSON.stringify(lock)}
-              lock={lock}
-              hideModal={hideModal}
-              showModal={showModal}
-              openInNewWindow={openInNewWindow}
-            />
-          ))}
-        </GlobalErrorConsumer>
-      </Locks>
-      <LockedFlag />
-    </Banner>
-  </FullPage>
-)
+  config: { isInIframe, requiredConfirmations },
+  transaction,
+  optimism,
+  smallBody,
+  bigBody,
+}) => {
+  const { postMessage } = usePostMessage()
+  useEffect(() => {
+    if (optimism.current && transaction) {
+      postMessage(POST_MESSAGE_GET_OPTIMISTIC)
+      smallBody()
+    } else {
+      // this branch should execute even if we are optimstic but have no transaction
+      postMessage(POST_MESSAGE_GET_PESSIMISTIC)
+      bigBody()
+    }
+  }, [optimism.current, transaction])
+  if (optimism.current && transaction) {
+    if (transaction.confirmations >= requiredConfirmations) {
+      return <ConfirmedFlag dismiss={hideModal} />
+    }
+    return <ConfirmingFlag transaction={transaction} lock={locks[0]} />
+  }
+  return (
+    <FullPage>
+      <Banner scrollPosition={scrollPosition} data-testid="paywall-banner">
+        <Headline id="Paywall_Headline">
+          {transaction && optimism.current === 0 && optimism.past === 1
+            ? 'Content unlock is still pending'
```

Comment:
Is that correct English? [I have no idea... but this seems confusing?]

Reply:
How about "content purchase is pending"?